dev-config.yaml

```diff
@@ -48,6 +53,34 @@ hub:
     requests:
       memory: 0
       cpu: 0
+  extraConfig:
+    # emit logs associated with our pytest tests
+    test_hub_existing_secret: |
+      try:
+          from z2jh import get_secret_value
+
+          print()
+          print("TEST LOGS: test_hub_existing_secret")
+          print(f"hub.existingSecret={get_config('hub.existingSecret')}")
+          print()
+          print(f"hub.db.type={get_config('hub.db.type')}")
+          print(f"MYSQL_PWD={os.environ.get('MYSQL_PWD', None)}")
+          print(f"PGPASSWORD={os.environ.get('PGPASSWORD', None)}")
+          print()
+          print(f"hub.services.test-hub-existing-secret.apiToken={get_secret_value('hub.services.test-hub-existing-secret.apiToken', None)}")
```

Comment:
get_secret_value doesn't take a second 'default' argument, leading to [this error](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/runs/1959618243#step:21:576):

> TypeError: get_secret_value() takes 1 positional argument but 2 were given

 But since you expect it to, maybe it should! That's certainly standard for `getters`.

After that, all LGTM!

Reply:
So to conlcude, the hub.extraConfig have a new entry in dev-config.yaml, which is meant to emit relevant information from the container enabling a test to decide if everything is as it should.

Since that is used for all tests, even the upgrade tests where we first install 0.11.1 (latest stable), and 0.11.1-n123.h1234567 (latest dev release), then it need to run successfully there also.

But, in the logic we make use of the function that was introduced after 0.11.1 and gained an optional parameter in this PR. To avoid issues for 0.11.1 I have an try/catch block for ImportErrors but not for any error such as passing a parameter that isn't defined.

So, no change is needed I think.

Reply:
Gotcha, thanks!